### ISSUE TITLE
Cannot run tests in VS after upgrade from 2.1.0

### ISSUE THREAD

**Issue by molszews** [machine/machine.specifications#328]
After upgrading package `Machine.Specifications.Runner.VisualStudio` from 2.1.0 to 2.2.0 (and 2.2.2) exception is thrown from the test runner:

```
------ Discover test started ------
Machine Specifications Visual Studio Test Adapter - Discovering Specifications.
Machine Specifications Visual Studio Test Adapter - Discovering...looking in xxTest.dll
Machine Specifications Visual Studio Test Adapter - Discovery Complete - 222 specifications in 1 of 1 assemblies scanned.
========== Discover test finished: 222 found (0:00:01.7600291) ==========
------ Run test started ------
Machine Specifications Visual Studio Test Adapter - Executing Specifications.
Machine Specifications Visual Studio Test Adapter - Executing tests in xxTest.dll
Machine Specifications Visual Studio Test Adapter - Error while executing specifications in assembly xxTest.dll - Could not load type 'Machine.VSTestAdapter.Execution.AppDomainExecutor' from assembly 'Machine.TestAdapter, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.
Complete on 1 assemblies 
No test is available in xxTest.dll. Make sure that installed test discoverers & executors, platform & framework version settings are appropriate and try again.
========== Run test finished: 0 run (0:00:00.8576154) ==========```

**Comment by ivanz** [comment]
How are you running the tests? VS IDE/vstest/visual studio team services?

**Comment by molszews** [comment]
original error log was from VS 2017 IDE, below is one from VSTS (2.1.0 works fine):
```
2017-03-31T09:09:59.6087343Z ##[section]Starting: Test Assemblies **\release\*Test.dll;-:**\obj\**
2017-03-31T09:09:59.6243528Z ==============================================================================
2017-03-31T09:09:59.6243528Z Task         : Visual Studio Test
2017-03-31T09:09:59.6243528Z Description  : Run tests with Visual Studio test runner
2017-03-31T09:09:59.6243528Z Version      : 2.0.19
2017-03-31T09:09:59.6243528Z Author       : Microsoft Corporation
2017-03-31T09:09:59.6243528Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkId=835764)
2017-03-31T09:09:59.6243528Z ==============================================================================
2017-03-31T09:10:08.8600378Z !**\obj\**
2017-03-31T09:10:08.8714990Z [command]powershell -NonInteractive -ExecutionPolicy Unrestricted -file C:\AzureBuildAgent\_work\_tasks\VSTest_ef087383-ee5e-42c7-9a53-ab56c98420f9\2.0.19\vs15Helper.ps1
2017-03-31T09:10:11.8130869Z <Objs Version="1.1.0.1" xmlns="http://schemas.microsoft.com/powershell/2004/04">
2017-03-31T09:10:11.8130869Z   <S>C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools</S>
2017-03-31T09:10:11.8130869Z </Objs>
2017-03-31T09:10:13.1154625Z [command]"C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\Common7\IDE\CommonExtensions\Microsoft\TestWindow\vstest.console.exe" C:\AzureBuildAgent\_work\44\s\src\xxTest.dll /logger:trx "/TestAdapterPath:\"C:\AzureBuildAgent\_work\44\s\""
2017-03-31T09:10:17.3839592Z Microsoft (R) Test Execution Command Line Tool Version 15.0.26228.0
2017-03-31T09:10:17.3839592Z Copyright (c) Microsoft Corporation.  All rights reserved.
2017-03-31T09:10:17.3839592Z 
2017-03-31T09:10:23.4988120Z Starting test execution, please wait...
2017-03-31T09:10:26.2132632Z Information: Machine Specifications Visual Studio Test Adapter - Executing Specifications.
2017-03-31T09:10:26.2132632Z 
2017-03-31T09:10:26.2132632Z Information: Machine Specifications Visual Studio Test Adapter - Executing tests in C:\AzureBuildAgent\_work\44\s\src\xxTest.dll
2017-03-31T09:10:26.2132632Z 
2017-03-31T09:10:28.3952727Z Error: Machine Specifications Visual Studio Test Adapter - Error while executing specifications in assembly C:\AzureBuildAgent\_work\44\s\src\xxTest.dll - Could not load type 'Machine.VSTestAdapter.Execution.AppDomainExecutor' from assembly 'Machine.TestAdapter, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.
2017-03-31T09:10:28.3952727Z Information: Complete on 1 assemblies 
2017-03-31T09:10:28.3952727Z 
2017-03-31T09:10:28.3952727Z 
2017-03-31T09:10:28.3952727Z Warning: No test is available in C:\AzureBuildAgent\_work\44\s\src\xxTest.dll. Make sure that installed test discoverers & executors, platform & framework version settings are appropriate and try again.
2017-03-31T09:10:28.3952727Z 
2017-03-31T09:10:28.3952727Z 
2017-03-31T09:10:28.3952727Z Information: Additionally, you can try specifying '/UseVsixExtensions' command if the test discoverer & executor is installed on the machine as vsix extensions and your installation supports vsix extensions. Example: vstest.console.exe myTests.dll /UseVsixExtensions:true
2017-03-31T09:10:28.3952727Z 
2017-03-31T09:10:28.3952727Z ##[warning]Vstest failed with error. Check logs for failures. There might be failed tests.
2017-03-31T09:10:28.3952727Z ##[error]Error: C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\Common7\IDE\CommonExtensions\Microsoft\TestWindow\vstest.console.exe failed with return code: 1
2017-03-31T09:10:28.3952727Z ##[warning]No results found to publish.
2017-03-31T09:10:28.3952727Z ##[error]Vstest exited with return code: 1.
2017-03-31T09:10:28.3952727Z ##[section]Finishing: Test Assemblies **\release\*Test.dll;-:**\obj\**
```

**Comment by ivanz** [comment]
Thanks a lot for the report. I believe I've fixed this in 2.2.3. I wasn't able to replicate the exact error you are seeing, but I believe to have fixed the root cause of discovery/run not working when the adapter is used through by the old vstest tooling for the "old" non-dotnet .csproj. Let me know if this works or doesn't 👍
